File: J_se_les12/HW/Core/Logic/Remove/RemoveV1.java
package Java.J_se_les12.HW.Core.Logic.Remove;

import Java.J_se_les12.HW.Core.Logic.Interfaces.IConsoleOutErrors;
import Java.J_se_les12.HW.Core.Logic.Interfaces.IConsoleOutSuc;
import Java.J_se_les12.HW.Core.Phonebook.Phonebook;

public class RemoveV1 extends RemoveBase {

    public RemoveV1(Phonebook phonebook, IConsoleOutSuc outSuc, IConsoleOutErrors error) {
        super(phonebook, outSuc, error);

    }

    @Override
    public void remove(int index) {
        if (this.phonebook.sizeOfBook() > 0) {
            this.phonebook.remove(index);
            outSuc.removeSuccess();
        }
    }
}

File: J_se_les13/HW/Core/Logic/Remove/RemoveV1.cs
using HW.Core.Phonebook;
using HW.Core.Logic.Interfaces;
namespace HW.Core.Logic.Remove
{
    public class RemoveV1 : RemoveBase
    {
        public RemoveV1(PhonebookV1 phonebook, IConsoleOutSuc outSuc, IConsoleOutError error)
        : base(phonebook, outSuc, error)
        {
        }
        public override void Remove(int index)
        {
            if (Phonebook.SizeOfBook() > 0)
            {
                Phonebook.Remove(index);
                OutSuc.RemoveSuccess();
            }
        }
    }
}